base/bucket_gocb_test.go

```diff
@@ -1752,6 +1753,77 @@ func TestXattrMutateDocAndXattr(t *testing.T) {
 
 }
 
+func TestGetXattr(t *testing.T) {
+	SkipXattrTestsIfNotEnabled(t)
+
+	testBucket := GetTestBucketOrPanic()
+	defer testBucket.Close()
+
+	defer SetUpTestLogging(LevelDebug, KeyAll)()
+
+	bucket, ok := testBucket.Bucket.(*CouchbaseBucketGoCB)
+	if !ok {
+		log.Printf("Can't cast to bucket")
+		return
+	}
+
+	//Doc 1
+	key1 := "DocExistsXattrExists"
+	val1 := make(map[string]interface{})
+	val1["type"] = key1
+	xattrName1 := "sync"
+	xattrVal1 := make(map[string]interface{})
+	xattrVal1["seq"] = float64(1)
+	xattrVal1["rev"] = "1-foo"
+
+	//Doc 2 - Tombstone
+	key2 := "TombstonedDocXattrExists"
+	val2 := make(map[string]interface{})
+	val2["type"] = key2
+	xattrName2 := "sync"
+	xattrVal2 := make(map[string]interface{})
+	xattrVal2["seq"] = float64(1)
+	xattrVal2["rev"] = "1-foo"
+
+	var err error
+
+	//Create w/ XATTR
+	cas := uint64(0)
+	cas, err = bucket.WriteCasWithXattr(key1, xattrName1, 0, cas, val1, xattrVal1)
+	if err != nil {
+		t.Errorf("Error doing WriteCasWithXattr: %+v", err)
+	}
+
+	var response map[string]interface{}
+
+	//Get Xattr From Existing Doc with Existing Xattr
+	_, err = testBucket.GetXattr(key1, xattrName1, &response)
+	assert.NoError(t, err)
+
+	assert.Equal(t, xattrVal1["seq"], response["seq"])
+	assert.Equal(t, xattrVal1["rev"], response["rev"])
+
+	//Get Xattr From Existing Doc With Non-Existent Xattr
+	_, err = testBucket.GetXattr(key1, "non-exist", &response)
+	assert.Error(t, err)
+
+	//Get Xattr From Non-Existent Doc With Non-Existent Xattr
+	_, err = testBucket.GetXattr("non-exist", "non-exist", &response)
+	assert.Error(t, err)
+
+	//Get Xattr From Tombstoned Doc With Existing Xattr
+	cas, err = bucket.WriteCasWithXattr(key2, xattrName2, 1, cas, val2, xattrVal2)
+	assert.NoError(t, err)
+	time.Sleep(2 * time.Second)
```

Comment:
I'm not actually sure why this test is passing as written - tombstones should only be retained for system xattrs, and system xattrs are identified by a leading underscore in the xattr name.  Maybe an expiry timing race?

You can avoid the expiry (and the sleep) by just creating the doc w/ a system xattr name (like "_sync"), and then issuing a bucket.Remove for the key.  The xattr will be retained with the tombstone, and then you can attempt retrieval.

Reply:
I have swapped it out for this method.